gomanifest/build.go

```diff
@@ -0,0 +1,234 @@
+type GoListCmdInterface interface {
+	Run() (string, error)
+}
+
+// GoList ... Structure that handle go list data and extract required packages.
+type GoList struct {
+	Command GoListCmdInterface
+}
+
+// Get ... Get deps data through go list deps command and converts json into objects.
+func (goList *GoList) Get() int {
+	output, err := goList.Command.Run()
+
+	if err != nil {
+		log.Println("ERROR :: Command `go list -json -deps ./...` failed, resolve project build errors.", err)
+		return -1
+	}
+
+	goListDepsData := string(output)
+	goListDepsData = `{"Packages": [` + strings.ReplaceAll(goListDepsData, "}\n{", "},\n{") + "]}"
+
+	var goPackages GoPackages
+	json.Unmarshal([]byte(goListDepsData), &goPackages)
+	log.Println("Packages in deps:", len(goPackages.Packages))
+
+	// Preprocess and remove all standard packages.
+	for i := 0; i < len(goPackages.Packages); i++ {
+		// Exclude standard packages
+		if goPackages.Packages[i].Standard == false {
+			depsPackages[goPackages.Packages[i].ImportPath] = goPackages.Packages[i]
+		}
+	}
+	log.Println("Filter package count:", len(depsPackages))
+
+	return 0
+}
+
+// getTransitives ... Returns a clean list of deps
+func getTransitives(deps []string) []ManifestTransitiveDeps {
+	var manifestDependencies = make([]ManifestTransitiveDeps, 0)
+	for i := 0; i < len(deps); i++ {
+		if depPackage, ok := depsPackages[deps[i]]; ok {
+			if depPackage.Module.Main == false {
+				manifestDependencies = append(manifestDependencies, ManifestTransitiveDeps{
+					getPackageName(depPackage),
+					transformVersion(depPackage.Module.Version),
+				})
+			}
+		}
+	}
+	return manifestDependencies
+}
+
+// buildManifest ... Build direct & transitive dependencies.
+func buildManifest(manifestFilePath string) {
+	var manifest Manifest = Manifest{manifestVersion, "", make([]MainfestDirectDeps, 0)}
+
+	// Get direct imports from current source.
+	var sourceImports = make(map[string]bool, 0)
+	for _, pckg := range depsPackages {
+		// Include only packages with project ROOT
+		if pckg.Module.Main == true {
+			// Set main module if not set.
+			if manifest.Main == "" {
+				manifest.Main = pckg.Module.Path
+			}
+
+			// Added imports as direct dependencies
+			for _, imp := range pckg.Imports {
+				// Add imports if not added yet.
+				if _, ok := sourceImports[imp]; !ok {
+					// Added imports that are non-standard (or present in deps packages) and
+					// which are having main module as 'false'
+					if depPackage, ok := depsPackages[imp]; ok {
+						if depPackage.Module.Main == false {
+							manifest.Packages = append(manifest.Packages,
+								MainfestDirectDeps{
+									getPackageName(depPackage),
+									transformVersion(depPackage.Module.Version),
+									getTransitives(depPackage.Deps),
+								})
+						}
+					}
+
+					sourceImports[imp] = true
+				}
+			}
+		}
+	}
+	log.Println("Source code imports:", len(sourceImports))
+
```

Comment:
This function shall return only the Manifest struct, JSON marshalling and Writing to file could be separated out for the sake of SRP rule.

Reply:
+1

Reply:
Build and save functionality is separated now.